cosmovisor/cmd/cosmovisor/cmd/help.go

```diff
@@ -0,0 +1,101 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/cosmovisor"
+	"os"
+)
+
+// ShouldGiveHelp checks the env and provided args to see if help is needed or being requested.
+// Help is needed if the os.Getenv(EnvName) env var isn't set.
+// Help is requested if the first arg is "help" or "--help"; or the only arg is "-h"
+func ShouldGiveHelp(args []string) bool {
+	if len(os.Getenv(cosmovisor.EnvName)) == 0 {
+		return true
+	}
+	if len(args) == 0 {
+		return false
+	}
+	return args[0] == "help" || args[0] == "--help" || (len(args) == 1 && args[0] == "-h")
```

Comment:
why do we have a special case for `"-h"`?

Reply:
Since `cosmovisor` is a wrapper for another application, I was scared that the other application might define a `-h` that doesn't mean `--help`.

However, I dug through the cobra stuff and found that `--help` and `-h` are added by default. The `help` subcommand is added by default too. Since the Cosmos SDK uses cobra, and `cosmovisor` is designed for apps that implement the Cosmos SDK, I now feel that it's a safe assumption that the `-h` flag will actually be a short form of `--help`.

If any arg is `-h` or `--help` then we should give help. If the first argument is `help`, then we should give help as well.

I limited `help` to only the first argument because, anywhere else (without the dashes), it's either a sub-command, an option value, or an illegal argument. All three are clearly meant for the other application, and in the case of "an option value", it might not actually be requesting help. As such, I didn't want this to drop them into the 2-second timeout, no monitoring version of running the command; doing so might make it impossible for someone to use `cosmovisor` as needed.

I guess, alternatively, I could undo the changes to `process.go` and always have it do the full setup and running of the command, even if we think it might just be for help. And the more I think about that, the more I like it.

Reply:
That last thought grew on me enough that I did it.

If any arg is `help`, `--help`, or `-h`, then help is printed. Then, if the config is valid, the DetailString() is printed and the configured command is run with the provided arguments the same way it would if on of those args weren't present. If the config is not valid, the config errors are printed and execution is stopped.